source/simulator/core.cc

```diff
@@ -726,14 +722,15 @@ namespace aspect
             VectorTools::interpolate_boundary_values (*mapping,
                                                       dof_handler,
                                                       *p,
-                                                      VectorFunctionFromScalarFunctionObject<dim>(std_cxx11::bind (&BoundaryTemperature::Manager<dim>::boundary_temperature,
-                                                          std_cxx11::cref(boundary_temperature_manager),
-                                                          *p,
-                                                          std_cxx11::_1),
-                                                          introspection.component_masks.temperature.first_selected_component(),
-                                                          introspection.n_components),
-                                                      current_constraints,
-                                                      introspection.component_masks.temperature);
+                                                      VectorFunctionFromScalarFunctionObject<dim>(
+                                                        [&] (const Point<dim> &x) -> double
+            {
+              return boundary_temperature_manager.boundary_temperature(*p, x);
+            },
+            introspection.component_masks.temperature.first_selected_component(),
```

Comment:
the remainder of the indentation is broken here, and since it's over several lines, I think it would be nice if you broke out the declaration of the `VectorFunctionFromScalarFunctionObject` object and made it a named object to be passed as argument here.